scripts/calc_flux_points.py

```diff
@@ -1,17 +1,26 @@
 from argparse import ArgumentParser
 
+from astropy.time import Time
 from gammapy.analysis import Analysis, AnalysisConfig
 from gammapy.datasets import Datasets
 
-parser = ArgumentParser()
-parser.add_argument("-c", "--config", required=True)
-parser.add_argument("--dataset-path", required=True)
-parser.add_argument("--best-model-path", required=True)
-parser.add_argument("-o", "--output", required=True)
-args = parser.parse_args()
 
+def select_timeframe(datasets, t_start, t_stop):
+    if t_start or t_stop:
+        t_start = Time(t_start, format="mjd") if t_start else datasets.gti.time_start[0]
+        t_stop = Time(t_stop, format="mjd") if t_stop else datasets.gti.time_stop[-1]
```

Comment:
This `if` clause makes no sense, you check both on their own again.

Reply:
Selecting based on time makes no sense if no range is given. This checks that not both are none, which is useful I think.
I did however end up with a robust implementation, that would then select all datasets, so you could argue, that we could skip the `if`

Reply:
Now it looks like a special case, although it isn't

Reply:
But it is?
We use the same script for the normal analysis